BeardedSpice/runningSBApplication.m

```diff
@@ -117,12 +117,122 @@ - (void)makeKeyFrontmostWindow{
 }
 
 /////////////////////////////////////////////////////////////////////////
-#pragma mark Private methods
+#pragma mark Supporting actions in application menubar
+
+- (NSString *)menuBarItemNameForIndexPath:(NSIndexPath *)indexPath {
+    
+    AXUIElementRef menuItem = [self copyMenuBarItemForIndexPath:indexPath];
+    
+    NSString *name;
+    if (menuItem) {
+        
+        CFTypeRef title;
+        name = (AXUIElementCopyAttributeValue(menuItem, (CFStringRef) NSAccessibilityTitleAttribute, (CFTypeRef *)&title) == kAXErrorSuccess ?
+                (NSString *)CFBridgingRelease(title): nil);
+        
+        CFRelease(menuItem);
+    }
+    
+    return name;
+}
+
+- (BOOL)pressMenuBarItemForIndexPath:(NSIndexPath *)indexPath {
+    
+    AXUIElementRef menuItem = [self copyMenuBarItemForIndexPath:indexPath];
+    
+    BOOL result = NO;
+    if (menuItem) {
+        
+        result = (AXUIElementPerformAction(menuItem, (CFStringRef)NSAccessibilityPressAction) == kAXErrorSuccess);
+        
+        CFRelease(menuItem);
+    }
+    
+    return result;
+}
+
 /////////////////////////////////////////////////////////////////////////
+#pragma mark Private methods
 
 - (NSRunningApplication *)runningAppication{
     NSArray *appArray = [NSRunningApplication runningApplicationsWithBundleIdentifier:self.bundleIdentifier];
     return [appArray firstObject];
 }
 
+- (AXUIElementRef)copyMenuBarItemForIndexPath:(NSIndexPath *)indexPath{
+    
+    if (! indexPath.length) {
+        return nil;
+    }
+
+    AXUIElementRef item = nil;
+    AXUIElementRef ref = AXUIElementCreateApplication(self.processIdentifier);
+    
+    if (ref) {
+        
```

Comment:
I put so many blank lines, since it is convenient for me to read the text. I usually put empty line after the beginning of the block of code (after '{'). But it is not important. :)